builder/azure/arm/step_capture_image.go

```diff
@@ -70,6 +68,18 @@ func (s *StepCaptureImage) captureImage(ctx context.Context, vmId hashiVMSDK.Vir
 	return nil
 }
 
+func (s *StepCaptureImage) getVMID(ctx context.Context, vmId hashiVMSDK.VirtualMachineId) (string, error) {
+	vmResponse, err := s.client.VirtualMachinesClient.Get(ctx, vmId, hashiVMSDK.DefaultGetOperationOptions())
+	if err != nil {
+		return "", err
+	}
+	if vmResponse.Model != nil {
+		vmId := vmResponse.Model.Properties.VMId
```

Comment:
Also, don't know how likely it is to happen, but can `Properties` be `nil` here? If so this will crash upon dereferencing it.

Reply:
TBH I think I need to standardize this across the plugin, null checking returned Models/properties, so I'd like to push off figuring that out to a later PR